File: my-submissions/2106-find-greatest-common-divisor-of-array/solution.java
class Solution {

    private int gcd(int a, int b) {
        while(b != 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
    public int findGCD(int[] nums) {
        Arrays.sort(nums); 

        int smallest = nums[0];
        int largest = nums[nums.length - 1];

       // System.out.println(nums[0] + ", " + nums[nums.length - 1]);

        return gcd(smallest, largest);
        
    }
}

File: my-submissions/2106-find-greatest-common-divisor-of-array/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findGCD = function(nums) {

    function gcd(a, b) {
        while(b != 0) {
            let temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }

    const sorted = nums.sort((a, b) => a - b);

    // console.log("sorted =>", sorted);

    const smallest = sorted[0];
    const largest = sorted[sorted.length - 1];

    // console.log("smallest, largest =>", smallest, largest);

    return gcd(smallest, largest)
    
};
